src/session/session_compact.c

```diff
@@ -334,6 +334,32 @@ __wt_session_compact(WT_SESSION *wt_session, const char *uri, const char *config
     session = (WT_SESSION_IMPL *)wt_session;
     SESSION_API_CALL(session, compact, config, cfg);
 
+    /* Trigger the background server thread and toggle connection statistic. */
+    if ((ret = __wt_config_getones(session, config, "background", &cval) == 0)) {
+        if (uri != NULL)
+            WT_ERR_MSG(session, EINVAL, "Background compaction does not work on specific URIs.");
+
+        if (!cval.val) {
+            WT_ERR_NOTFOUND_OK(__wt_config_getones(session, config, "timeout", &cval), true);
+            if (ret == 0)
+                WT_ERR_MSG(session, EINVAL,
+                  "timeout configuration cannot be set when disabling the background compaction "
+                  "server.");
+
+            WT_ERR_NOTFOUND_OK(
+              __wt_config_getones(session, config, "free_space_target", &cval), true);
+            if (ret == 0)
+                WT_ERR_MSG(session, EINVAL,
+                  "free_space_target configuration cannot be set when disabling the background "
+                  "compaction server.");
+        }
+
+        WT_ERR(__wt_compact_signal(session, config));
+
+        return (0);
```

Comment:
Do we need any code under the `err` label this is still executed when everything goes well such as `API_END_RET_NOTFOUND_MAP(session, ret);`?

Reply:
That's relevant for when we actually try to compact a table. It will be executed later when the background compaction thread calls this function. The only check that I think might be worthwhile is the in memory one. But in that case we probably shouldn't even have a background compaction thread. Should we improve how we handle that case now? Or something to consider later?

Reply:
Good point on the `API_END_RET_NOTFOUND_MAP(session, ret);` I'll fix this.